api/src/main/java/org/apache/iceberg/data/Record.java

```diff
@@ -35,4 +36,26 @@
   Record copy();
 
   Record copy(Map<String, Object> overwriteValues);
+
+  default Record copy(String field, Object value) {
+    Map<String, Object> overwriteValues = Maps.newHashMapWithExpectedSize(1);
+    overwriteValues.put(field, value);
+    return copy(overwriteValues);
+  }
+
+  default Record copy(String field1, Object value1, String field2, Object value2) {
+    Map<String, Object> overwriteValues = Maps.newHashMapWithExpectedSize(1);
```

Comment:
Shouldn't this be 2?

Reply:
I looked into the code for this, it's actually interesting that 1 and 2 follow a relatively fast path while 3 will require doing some power of 2 estimation.

https://github.com/google/guava/blob/064fac35d71231aba35062d1965983ecd36b6873/guava/src/com/google/common/collect/Maps.java#L324-L328

Reply:
Yeah, I'll update this.